src/libstd/sys/common/io.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+use prelude::v1::*;
+use io;
+use io::ErrorKind;
+use io::Read;
+
+pub const DEFAULT_BUF_SIZE: usize = 64 * 1024;
+
+
+pub fn read_to_end_zeroed<R: Read + ?Sized>(r: &mut R, buf: &mut Vec<u8>) -> io::Result<usize> {
+    unsafe {read_to_end(r, buf, true) }
+}
+
+pub unsafe fn read_to_end_uninitialized<R: Read + ?Sized>(r: &mut R, buf: &mut Vec<u8>) -> io::Result<usize> {
```

Comment:
libstd should be smaller if there's only one instantiation of this function. It's just a random idea, not so important.